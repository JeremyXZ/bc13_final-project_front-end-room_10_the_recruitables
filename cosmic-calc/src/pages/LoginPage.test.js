import React from "react";
import "@testing-library/jest-dom";
import { screen, render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import {BrowserRouter, MemoryRouter, useLocation} from 'react-router-dom'

import Login from "./LoginPage";

describe('<LoginPage/> rendering and navigataing', () => {
    
    it("successfully renders the component, matches snapshot", () => {
        const { asFragment } = render(<Login />, {wrapper: BrowserRouter});
        expect(asFragment()).toMatchSnapshot();
      });

     
    it("verifies page content for default route", async () => {
        render(<Login />, {wrapper: BrowserRouter})              
        expect(screen.getByPlaceholderText(/email/i)).toBeInTheDocument()
        expect(screen.getByPlaceholderText(/password/i)).toBeInTheDocument()
        // expect(screen.getByRole('button', {name: /login/i})).toBeInTheDocument()   
        // expect(screen.getByTestId('loginButton').textContent).toBe("50 + 20 =");
    })

})

