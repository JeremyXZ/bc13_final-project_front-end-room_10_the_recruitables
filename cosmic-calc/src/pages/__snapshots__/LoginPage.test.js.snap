// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login/> rendering and navigataing successfully renders the component, matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Login"
  >
    <div>
      <div
        class="loginLogoDiv"
      >
        <img
          alt="logo"
          class="loginLogo"
          src="Logo.png"
        />
      </div>
      <form>
        <div
          class="inputLoginDiv"
        >
          <input
            class="loginInput"
            placeholder="email"
            type="email"
            value=""
          />
          <input
            class="passwordInput"
            placeholder="password"
            type="password"
            value=""
          />
        </div>
        <div
          class="loginButtonDiv"
        >
          <button
            class="loginButton"
            type="submit"
          >
            Login
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<LoginPage/> rendering and navigataing successfully renders the component, matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Login"
  >
    <div>
      <div
        class="loginLogoDiv"
      >
        <img
          alt="logo"
          class="loginLogo"
          src="Logo.png"
        />
      </div>
      <form>
        <div
          class="inputLoginDiv"
        >
          <input
            class="loginInput"
            placeholder="email"
            type="email"
            value=""
          />
          <input
            class="passwordInput"
            placeholder="password"
            type="password"
            value=""
          />
        </div>
        <div
          class="loginButtonDiv"
        >
          <button
            class="loginButton"
            type="submit"
          >
            Login
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
